File: lab2.kt
fun main() {
    // Задание №1.2
    println("Задание №1.2")
    val inputString = "abc1231de45f42" //abc1234def4256gh789ab999c14234def516gh789abc12314def56gh789
    var max = 0
    var numb = 0

    for (char in inputString) {
        val a = char.toInt()
        if (a in '0'.toInt()..'9'.toInt()) { // Проверка на цифру
            numb += a - '0'.toInt()
            println("${a - '0'.toInt()} $numb")
        } else {
            if (max < numb) {
                max = numb
            }
            numb = 0
        }
    }
    if (max < numb) {
        max = numb
    }
    println(max)
    //Задание №2.8
    println("Задание №2.8")
    val grad = mutableListOf(0) // Инициализация списка grad
    var steps = 0
    val leika_old = 10
    var leika_now = leika_old
    val str2 = "2 2 3 3 3 5"
    var a_f = 0

    // Обработка строки для извлечения чисел
    for (char in str2) {
        val a = char.toInt()
        if (a in '0'.toInt()..'9'.toInt()) { // Проверка на цифру
            a_f = a_f * 10 + (a - '0'.toInt())
        } else if (a == ' '.toInt()) { // Обработка пробела
            if (a_f > 0) {
                grad.add(a_f)
                a_f = 0
            }
        }
    }
    if (a_f > 0) {
        grad.add(a_f)
    }

    // Вывод значений grad
    println(grad.joinToString(" "))

    // Логика для подсчета шагов
    var i = 0
    while (i < grad.size - 1) {
        if (leika_now >= grad[i + 1]) {
            i++
            leika_now -= grad[i]
            grad[i] = 0
            steps++
        } else {
            while (i > 0) {
                steps++
                i--
            }
            leika_now = leika_old
        }
    }

    // Вывод значений grad после обработки
    println(grad.joinToString(" "))
    println(steps)
    // Задание №3.18
    println("Задание №3.18")
    println("Кол-во чисел: ")
    val n = readLine()!!.toInt()
    var num: Int
    var numR: Int
    var numTex: Int

    for (j in 0 until n) {
        println("Введите число: ")
        num = readLine()!!.toInt()
        numTex = num
        numR = 0

        while (numTex != 0) {
            numR = numR * 10 + numTex % 10
            numTex /= 10
        }

        if (num > numR) {
            println(num - numR)
        } else {
            println(numR - num)
        }
    }
}

File: lab2.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        // Задание №1.2
        Console.WriteLine("задание №1.2");
        string inputString = "abc1231de45f42"; // введённая строка
        int max = 0;
        int numb = 0;

        for (int i = 0; i < inputString.Length; i++)
        {
            int a = (int)inputString[i]; // а принимает значения из строки и преобразует их в целое число
            if (a >= '0' && a <= '9') // Проверка на цифру
            {
                numb += a - '0';
            }
            else
            {
                if (max < numb)
                {
                    max = numb;
                }
                numb = 0;
            }
        }
        if (max < numb)
        {
            max = numb;
        }
        Console.WriteLine(max);

        // Задание №2.8
        Console.WriteLine("задание №2.8");
        List<int> grad = new List<int> { 0 }; // список чисел аналог вектора в c++
        int steps = 0;
        int leikaOld = 10;
        int leikaNow = leikaOld;
        string str2 = "2 2 3 3 3 5";
        int aF = 0; // переменная для заполнения grad

        for (int i = 0; i < str2.Length; i++)
        {
            int a = (int)str2[i];
            if (a >= '0' && a <= '9') // Проверка на цифру
            {
                aF = aF * 10 + (a - '0');
            }
            else
            {
                grad.Add(aF);
                aF = 0;
            }
        }
        grad.Add(aF); // Добавляем последнее число
        aF = 0;

        // Вывод значений grad
        Console.WriteLine(string.Join(" ", grad));
        // шагаем по грядке пока не достигнем конца, если воды хватает поливаем и идём вперёд, иначе возвращаемся в конец
        int j = 0;
        while (j < grad.Count - 1)
        {
            if (leikaOld < grad[j+1])
            {
                Console.Write("Ошибка");
                return;
            }
            if (leikaNow >= grad[j + 1])
            {
                j++;
                leikaNow -= grad[j];
                grad[j] = 0;
                steps++;
            }
            else
            {
                while (j > 0)
                {
                    steps += 1;
                    j--;
                }
                leikaNow = leikaOld;
            }
        }

        // Вывод значений 
        Console.WriteLine(steps);

        // Задание №3.18
        Console.WriteLine("задание №3.18");
        Console.Write("Кол-во чисел: ");
        int n = int.Parse(Console.ReadLine()); //ввoд из консоли преобразованый в целое число
        int num;
        int numR;
        int numTex;

        for (int k = 0; k < n; k++)
        {
            Console.Write("Введите число: ");
            num = int.Parse(Console.ReadLine());
            numTex = num;
            numR = 0;

            while (numTex != 0) // переворот числа
            {
                numR = numR * 10 + (numTex % 10);
                numTex /= 10;
            }

            if (num > numR) // вывод модуля числа
            {
                Console.WriteLine(num - numR);
            }
            else
            {
                Console.WriteLine(numR - num);
            }
        }
    }